pkg/common/resources/resources.go

```diff
@@ -884,6 +909,29 @@ func ComponentWiseMinPermissive(left, right *Resource) *Resource {
 	return out
 }
 
+// MergeIfNotPresent Returns a new Resource by merging resource type values present in right with left
+// only if resource type not present in left.
+// If either Resource passed in is nil the other Resource is returned
+// If a Resource type is missing from one of the Resource, it is considered empty and the quantity from the other Resource is returned
+func MergeIfNotPresent(left, right *Resource) *Resource {
+	if right == nil && left == nil {
+		return nil
+	}
+	if left == nil {
+		return right.Clone()
+	}
+	if right == nil {
+		return left.Clone()
+	}
+	out := left
```

Comment:
I think this should be `out := left.Clone()`, otherwise you are mutating the original resource. This seems wrong, since you're returning a new one.

Reply:
This still needs to be updated. This should never update left, and without a clone it will.